core/src/avm2/object/netstream_object.rs

```diff
@@ -15,15 +15,16 @@ pub fn netstream_allocator<'gc>(
     activation: &mut Activation<'_, 'gc>,
 ) -> Result<Object<'gc>, Error<'gc>> {
     let base = ScriptObjectData::new(class);
-
-    Ok(NetStreamObject(GcCell::allocate(
+    let ns = NetStream::new(activation.context.gc_context, None);
+    let this: Object<'gc> = NetStreamObject(GcCell::allocate(
         activation.context.gc_context,
-        NetStreamObjectData {
-            base,
-            ns: NetStream::new(activation.context.gc_context),
-        },
+        NetStreamObjectData { base, ns },
     ))
-    .into())
+    .into();
+
+    ns.set_avm_object(activation.context.gc_context, this.into());
```

Comment:
Nit: I do not know if I like the cyclic reference between the `Object` and `NetStream`, but it might be fine, and alternatives, like a callback, might be overall worse.